norman_core/clients/http_client.py

```diff
@@ -32,14 +33,20 @@ def __init__(self, base_url: Optional[str] = None, timeout: Optional[float] = No
         }
 
     async def open(self) -> None:
-        await self.close()
-        self._client = httpx.AsyncClient(
-            base_url=self._base_url,
-            timeout=self._timeout
-        )
+        if self._open_count == 0:
```

Comment:
Bug:

1) if the open count is zero, closing will result in an exception
2) as the connection is now reentrant, closing an existing connection will break the track on your reentrance counter, resulting in premature closing.

It is expected that callers manage their connections (refer to our original "guiding principle" ruleset), and this includes not to open the same connections twice.